lib/testers/rule-tester.js

```diff
@@ -365,20 +362,6 @@ class RuleTester {
                 }
             }));
 
-            if (schema) {
-                ajv.validateSchema(schema);
-
-                if (ajv.errors) {
-                    const errors = ajv.errors.map(error => {
-                        const field = error.dataPath[0] === "." ? error.dataPath.slice(1) : error.dataPath;
-
-                        return `\t${field}: ${error.message}`;
-                    }).join("\n");
-
-                    throw new Error([`Schema for rule ${ruleName} is invalid:`, errors]);
-                }
-            }
-
             validator.validate(config, "rule-tester", ruleMap.get.bind(ruleMap), new Environments());
```

Comment:
after the change, the error come from `validator.validate()`, which is a little different than before.

Reply:
I'm okay with the different format-- what I'm concerned about is that the actual validation error is different. (One mentioned matching an anyOf schema, the other mentioned an invalid enum value.) Why did the actual validation error change?

Reply:
good catch! I guess I've found the difference: `ajv.validateSchema()` was explicitly called, but after the change it was no longer called, and the error was thrown in `ajv.compile()`. so I am wondering if we can remove it.

it seems a good idea to add a new rule to `eslint-plugin-eslint-plugin`(https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin)

Reply:
Should we change this to explicitly call `validateSchema` once, rather than calling it for each test case?

Reply:
agreed! I enabled it in https://github.com/eslint/eslint/pull/10713/commits/21cfcff1e9edcf22021e64d3c81010890e566e28 , `ajv` has a cache by default